tf-build-tools/tf-maven-plugin/src/main/java/com/deliveredtechnologies/maven/terraform/TerraformUpload.java

```
@@ -0,0 +1,95 @@
+      String planFileName = planOutputFile.startsWith("s3") ? planOutputFile.split("/")[planOutputFile.split("/").length - 1] : planOutputFile;
+      logger.debug("checking for plan file " + planFileName);
+      String sse = properties.getProperty(TerraformUploadParams.sse.toString()) == null ? "AES:256" : properties.getProperty(TerraformUploadParams.sse.toString());
+      String kmsKeyId = properties.getProperty(TerraformUploadParams.kmsKeyId.toString());
+      if (planOutputFile.startsWith("s3")) {
+        response = String.format("Plan saved to %1$s", planOutputFile);
+        logger.debug("File Uploading to S3");
+        terraformUploadCli(planFileName, planOutputFile, sse, kmsKeyId, false);
+      }
+    }
+    return response;
+  }
+
+  /**
+   * Uploads objects to s3.
+   * @param body body of the content that need to be uploaded
+   * @param key fully qualified path for the s3 object to store or destination location of the file
+   * @param sse Server side encryption mechanism either of AES:256 or aws:kms; defaults to AES:256
+   * @param kmsKeyId KmsKeyId to encrypt the Objects.
+   * @param recursive extends s3 cp command with recursive--true if sets to true
+   *
+   */
+  protected void terraformUploadCli(String body, String key, String sse, String kmsKeyId, boolean recursive ) throws IOException {
+    try {
+      String recCmd = recursive ? "--recursive" : "";
+      if (sse.equals("AES:256")) {
+        logger.debug("uploading plan files to Amazon S3 with default encryption (SSE:256)");
+        executable.execute(String.format("aws s3 cp %1$s %2$s %3$s", body, key, recCmd));
+
+      } else {
+        logger.debug("uploading plan files to Amazon S3 with kms encryption");
+        executable.execute(String.format("aws s3 cp %1$s %2$s --sse %3$s --sse-kms-key-id %4$s %5$s", body, key, sse, kmsKeyId, recCmd));
+      }
+    } catch (InterruptedException e) {
+      e.printStackTrace();
```

printStackTrace() is not handling the exception. It would be better to just let it bubble up where the stack trace would also get printed.